gateway/metrics/add_metrics_test.go

```diff
@@ -0,0 +1,81 @@
+package metrics
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type testFetchQuery struct {
+	res *VectorQueryResponse
+	err error
+}
+
+func (f testFetchQuery) Fetch(query string) (*VectorQueryResponse, error) {
+	return f.res, f.err
+}
+
+func TestAddMetricsHandlerServeHTTPReturnsContentTypeOfPassedInHandlerWhenFetchSucceeds(t *testing.T) {
```

Comment:
I am not clear what these tests are doing. We always need to return Content-Type of `application/json`.

Reply:
These tests prove the issue described in the PR. They can be extended to cover more of the logic in the AddMetricsHandler function.

Reply:
I think the function needs to return `application/json` the whole time, unless it's an error as in an exception.